src/components/charts/LineChartMetric.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropType from 'prop-types';
+import {
+  LineChart,
+  Line,
+  XAxis,
+  YAxis,
+  CartesianGrid,
+  Legend,
+  Tooltip,
+} from "recharts";
+import { getColorFromGradient } from '../../utils/colors';
+const truncate = (str, n) => (str.length > n) ? str.substr(0, n-1) + '…' : str;
+
+const LineChartMetric = ({ width = 260, height = 140, series, data, showLegend }) => {
+  const mgBottom = showLegend ? 0 : -10;
+  
+  if (data && !Array.isArray(data[0])) {
+    series = [{ name: 'Line', data }];
+  }
+  
+  const lines = series.map((item, idx) => ({
+    key: item.name,
+    dataKey: `value`,
+    data: item.data,
+    name: truncate(item.name, 20),
+    color: getColorFromGradient('rgb(255,233,36)', 'rgb(242,11,11)', (idx + 1) / 5)
+  }));
+
+  if (showLegend) height += (lines.length * 15);
```

Comment:
This doesn't seem to have the desired effect:
<img width="299" alt="Screen Shot 2021-06-21 at 12 47 07 PM" src="https://user-images.githubusercontent.com/244952/122798849-0cb0c600-d28f-11eb-9aba-07ff4a8b6250.png">

The chart seems to take up a fixed percentage of the total height, instead of using what's left after the legend is laid out.

This might be solvable using Legend's `wrapperStyle` and some careful CSS.

Reply:
Some further information: going to comparison and back again makes this render a little more correctly. So this might be an issue with the chart not redoing its layout when its data changes.

Reply:
Other people having a similar issue with some suggested workarounds https://github.com/recharts/recharts/issues/511.